lib/sycamore/sycamore/query/execution/operations.py

```diff
@@ -127,12 +127,18 @@ def _get_text_for_summarize_data(
 
         # consolidates relevant properties to give to LLM
         if isinstance(result, DocSet):
-            for i, doc in enumerate(result.take(NUM_DOCS_GENERATE, **kwargs)):
+            done = False
+            # For query result caching in the executor, we need to consume the documents
+            # so that the materialized data is complete, even if they are not all included
+            # in the input prompt to the LLM.
+            for di, doc in enumerate(result.take_all()):
```

Comment:
This works at small scale, but will blow up memory at large scale.
Approving as this is NTSB only, but I suggest a TODO or something.